importers/ableroad.com/easy_importer.py

```diff
@@ -0,0 +1,226 @@
+    m = re.search(r'(?P<city>[a-zA-Z]+),\s*(?P<state>[a-zA-Z]+)', s)
+    if not m:
+        return ''
+    return m.group('state')
+
+def getpostcode(s):
+    if not len(s):
+        return ''
+    s = ' '.join(s)
+    m = re.search(r'(?P<city>[a-zA-Z]+),\s*(?P<state>[a-zA-Z]+)\s+(?P<post>(\d{5,5})|(\w{3}\s\w{3}))', s)
+    if not m:
+        return ''
+    return m.group('post')
+
+def getphone(s):
+    if len(s) < 1:
+        return ''
+    if not re.search(r'\d{2,}-\d+',s[-1]):
+        return ''
+    return s[-1]
+
+def extract_info(dombus):
+    row = []
+    name = dombus.xpath('.//a[@class="titlelink"]/text()')
+    name = getname(name)
+    row.append(name)
+
+    category = dombus.xpath('.//div[@class="category"]/text()')
+    category = getcategory(category)
+    row.append(category)
+
+    distance = dombus.xpath('.//div[@class="itemdistance"]/text()')
+    distance = getdistance(distance)
+    row.append(distance)
+
+    neighborhood = dombus.xpath('.//div[@class="neighborhood"]/text()')
+    neighborhood = getneighborhood(neighborhood)
+    row.append(neighborhood )
+
+    ableroadating = dombus.xpath('.//div[@class="searchableRating"]//div[@class="visually-hidden startableft"]/text()')
+    ableroadating = getableroad(ableroadating)
+    row.append(ableroadating)
+
+    yelprating = dombus.xpath('.//img[@class="yelprating"]/@alt')
+    yelprating = getyelp(yelprating)
+    row.append(yelprating)
+
+    address = dombus.xpath('.//address/text()')
+
+    street = getstreet(address)
+    row.append(street)
+
+    city = getcity(address)
+    row.append(city)
+
+    state = getstate(address)
+    row.append(state)
+
+    postcode = getpostcode(address)
+    row.append(postcode)
+
+    phone = getphone(address)
+    row.append(phone)
+
+    latlng = dombus.getprevious().text #lat lng is hidden in javascript
+    latlng = getlatlng(latlng)
+    row = row + latlng
+    return row
+
+def write_csv(rows):
+    csv_filename = 'export.csv'
+    if os.path.exists(csv_filename):
+        csv_file = open(csv_filename,'a')
+        writer = csv.writer(csv_file, delimiter=',',quotechar='"', quoting=csv.QUOTE_ALL)
+    else:
+        csv_file = open(csv_filename,'w')
+        writer = csv.writer(csv_file, delimiter=',',quotechar='"', quoting=csv.QUOTE_ALL)
+        title = ['name','category','distance','neighborhood','ablerating','yelprating','street','city',
+                'state','postcode','phone','lat','lng']
+        writer.writerow(title)
+    for row in rows:
+        writer.writerow(row)
+    csv_file.close()
+    logging.info('csv file has been written!')
+
+def download_if_not(url, islocalfile):
+    pagecontent = ''
+
+    if islocalfile and os.path.exists('tmp.html'):
+        f = open('tmp.html','r')
+        pagecontent = f.read()
+    else:
+        page = requests.get(url)
+        f = open('tmp.html','w')
+        f.write(page.content)
+        pagecontent = page.content
+
+    f.close()
+    return pagecontent
+
+def gen_url():
```

Comment:
Take the string like ['windsor, ontario'] as a parameter to gen_url.
```
def gen_url(locations):
```

Get the location(s) from command line arguments so the importer can generate a CSV for any specified location.  This will help us quickly add data from ableroad that might come from other cities.